PartWebApp2/Controllers/PartiesController.cs

```diff
@@ -199,32 +228,62 @@ public IActionResult Create()
             ViewData["Clubs"] = new SelectList(_context.Set<Club>(), "Id", "Name");
             ViewData["Areas"] = new SelectList(_context.Set<Area>(), "Id", "Type");
             ViewData["PerformersId"] = new SelectList(_context.Set<Performer>(), "Id", "SpotifyId");
-           
+
             return View();
         }
 
-        // POST: Parties/Create
-        // To protect from overposting attacks, enable the specific properties you want to bind to.
-        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+
         [HttpPost]
         [ValidateAntiForgeryToken]
         [Authorize(Roles = "Admin, producer")]
         public async Task<IActionResult> Create([Bind("Id,name,eventDate,minimalAge,startTime,genreId,areaId,clubId,ProducerId,maxCapacity,price,ticketsPurchased")] Party party, string imageUrl, List<string> performersId)
         {
             if (ModelState.IsValid)
             {
-                var producerId = HttpContext?.User?.FindFirstValue(ClaimTypes.NameIdentifier);
-                party.ProducerId = Int32.Parse(producerId);
+                party.ProducerId = findCurrentUserId();
+
                 party.ticketsPurchased = 0;
                 party.performers = new List<Performer>();
                 _partiesService.addPerformersToParty(party, performersId);
+                imageUrl = _partiesService.defaultImageIfIsNull(imageUrl);
```

Comment:
move that function to the service flow because there are both declared at the service